### DISCUSSION TITLE
ipxe-master compile time

### DISCUSSION THREAD

**Discussion by tgolightly** [ipxe/ipxe#1371] [Q&A] [answered]
Greetings,

I am new here.  Please go easy on me.  I have not compiled anything since the 90's.  I have been searching around this site as welll as across the internet and now I am here. Trying to compile ipxe to troubleshoot pxe booting 2012 and older MacBookPro's.  I run the command:

make bin-x86-efi/ipxe.efi DEBUG=tg3 EMBED=testscript.ipxe

testscript contains:
#!ipxe
dhcp
ifstat
shell

I have started compiling and left it go overnight.  It was still going (at least 4 hours).  How long does this normally take? Is there a way to not compile every known nic driver? I probably missed something along the way here.

Tia

**Comment by NiKiZe** [comment]
Sounds like you have some local modifications in config that is incompatible.
Do try a clean checkout, a make with -j8 or similar doesn't take more than one or a few minutes on "normal desktop cpu"

Every .c is always compiled, but not all code is linked into the final binary.

If you have issues with tg3, see existing topics on that (no known fix afaik)

**Comment by mcb30** [comment, marked as answer]
> make bin-x86-efi/ipxe.efi DEBUG=tg3 

This is not a valid build target. I think you might have meant to use `bin-x86_64-efi/ipxe.efi`.

**Reply by tgolightly** [reply]
tx for your reply.  Yes I am new to this.  I havn't compiled anything (successfully) since 1997.  I am still working on something in this area.  but for now I am not re-compiling ipxe for my fog server at work.  Still playing around at home though.
